lib/engine.py

```diff
@@ -425,17 +418,17 @@ def download_files_from_archive(self, url, filenames, filetype="zip",
             else:
                 self.create_raw_data_dir()
                 if not downloaded:
-                    self.download_file(url, archivename, clean_line_endings=False)
+                    self.download_file(url, archivename)
                     downloaded = True
 
                 if filetype == 'zip':
                     archive = zipfile.ZipFile(archivename)
-                    open_archive_file = archive.open(filename)
+                    open_archive_file = archive.open(filename, 'rU')
                 elif filetype == 'gz':
                     # gzip archives can only contain a single file
-                    open_archive_file = gzip.open(archivename)
+                    open_archive_file = gzip.open(archivename, 'rU')
```

Comment:
so basically when we read a file with universal newlines ''rU', we are reading it in as text. Python maps all of the characters \r, \n and \r\n to \n. However the retriever also handles binary data and this may be a bug.